### PR TITLE
Add installer hash, and sdk version from `sdk/*/.version` file

### PR DESCRIPTION
**Pull request by radical**
The sdk version is useful if `DOTNET_VERSION` does not reflect the real
version.

### CODE REVIEW

**Comment by radical** [comment]
Would it more useful to have the sdk hash instead of the installer? Or both? This info will be visible in the powerbi reports.

**Comment by radical** [comment]
Test run - https://dev.azure.com/dnceng/internal/_build/results?buildId=2277611 .

File: src/tools/Reporting/Reporting/Reporter.cs [hunk 1]
```diff
@@ -103,20 +104,57 @@ private void Init()
...
             else if(entry.Key.ToString().StartsWith("PERFLAB_DATA_", true, CultureInfo.InvariantCulture))
             {
                 build.AdditionalData[entry.Key.ToString().Substring("PERFLAB_DATA_".Length)] = entry.Value.ToString();
             }
-        }            
+        }
+
+        (string installerHash, string sdkVersion) GetDotNetVersionInfo(string dotnetRoot)
+        {
+            if (Path.Combine(dotnetRoot, "sdk") is not string sdkRootPath || !Directory.Exists(sdkRootPath))
```
[the remark this answers is not in the document]
Comment (radical) [reply src/tools/Reporting/Reporting/Reporter.cs:127]: I was trying to minimize nesting.

[the remark this answers is not in the document]
Comment (radical) [reply src/tools/Reporting/Reporting/Reporter.cs]: Fixed.

**Review by cincuranet — APPROVED** [review]

**Comment by radical** [comment]
The failures should be unrelated. @cincuranet if that looks correct, then you could you please merge?

**Pull request merged**